packages/material-ui-lab/src/Slider/Slider.js

```diff
@@ -211,426 +59,626 @@ function getMousePosition(event, touchId) {
-              disableRipple
-              onBlur={this.handleBlur}
-              onKeyDown={this.handleKeyDown}
-              onTouchStartCapture={this.handleTouchStart}
-              onTouchMove={this.handleTouchMove}
-              onFocusVisible={this.handleFocus}
-              role="slider"
+  return (
+    <Component
+      ref={handleRef}
+      className={clsx(
+        classes.root,
+        {
+          [classes.disabled]: disabled,
+          [classes.marked]: marks.length > 0 && marks.some(mark => mark.label),
+          [classes.vertical]: orientation === 'vertical',
+          [classes.rtl]: theme.direction === 'rtl',
+        },
+        className,
+      )}
+      onMouseDown={handleMouseDown}
+      {...other}
+    >
+      <span className={classes.rail} />
+      <span className={classes.track} style={trackStyle} />
+      <input value={values.join(',')} name={name} type="hidden" />
+      {marks.map(mark => {
+        const percent = valueToPercent(mark.value, min, max);
+        const style = axisProps[axis].offset(percent);
+
+        const active = range
+          ? percent >= values[0] && percent <= values[values.length - 1]
+          : percent <= values[0];
+
+        return (
+          <React.Fragment key={mark.value}>
+            <span
+              style={style}
+              className={clsx(classes.mark, {
+                [classes.markActive]: active,
+              })}
+            />
+            <span
+              style={style}
+              className={clsx(classes.markLabel, {
+                [classes.markLabelActive]: active,
+              })}
             >
-              {ThumbIcon}
-            </ButtonBase>
-          </div>
-          <div className={trackAfterClasses} style={inlineTrackAfterStyles} />
-        </div>
-      </Component>
-    );
-  }
-}
+              {mark.label}
+            </span>
+          </React.Fragment>
+        );
+      })}
+      {values.map((value, index) => {
+        const percent = valueToPercent(value, min, max);
+        const style = axisProps[axis].offset(percent);
+
+        return (
+          <ThumbLabelComponent
+            key={index}
+            thumbLabelFormat={thumbLabelFormat}
+            thumbLabelDisplay={thumbLabelDisplay}
+            className={classes.thumbLabel}
+            value={value}
+            index={index}
+            disabled={disabled}
+          >
+            <span
+              className={classes.thumb}
+              tabIndex={disabled ? null : 0}
+              role="slider"
+              style={style}
+              data-index={index}
+              aria-label={ariaLabel}
+              aria-labelledby={ariaLabelledby}
+              aria-orientation={orientation}
+              aria-valuemax={max}
+              aria-valuemin={min}
+              aria-valuenow={value}
+              aria-valuetext={
+                typeof ariaValuetext === 'function' ? ariaValuetext(value, index) : ariaValuetext
```

Comment:
We should not overload standardized html attributes. This can be very confusing from a user perspective. A specialized prop should be sufficient e.g. `getAriaValueText`. We should warn if `aria-valuetext` is provided for multi value sliders and warn if it's not provided for single value sliders and not getAriaValueText is provided.

Reply:
Yeah, I agree, I'm changing it.